### PR TITLE
Export typed event emitter key types

### LINKED ISSUE #3506: Expose `EmittedEvents` subtypes
Currently it is a compound of these:

```
ClientEvent | RoomEvents | RoomStateEvents | CryptoEvents | MatrixEventEvents | RoomMemberEvents | UserEvents | CallEvent | Incoming | Incoming | Outgoing | Ended | Participants | SessionLoggedOut | NoConsent | BeaconEvent
```

But few of them are exported and many like `CallEvent`, `Incoming` and `RoomEvent` are not exported and therefore cannot be used.  And we have to hardcode their values in our apps rather than use these enums which is the right way.

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/matrix-org/matrix-js-sdk/issues/3506

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Export typed event emitter key types ([\#3597](https://github.com/matrix-org/matrix-js-sdk/pull/3597)). Fixes #3506.

### CODE REVIEW

File: src/matrix.ts [hunk 1]
```diff
@@ -63,9 +63,23 @@ export * as SecretStorage from "./secret-storage";
...
+    GroupCallEvent,
+    GroupCallIntent,
+    GroupCallState,
+    GroupCallType,
+    GroupCallStatsReportEvent,
+} from "./webrtc/groupCall";
 export type { GroupCall } from "./webrtc/groupCall";
 export { CryptoEvent } from "./crypto";
+export { SlidingSyncEvent } from "./sliding-sync";
+export { VerificationRequestEvent, VerifierEvent } from "./crypto-api/verification";
```
Comment (richvdh) [inline src/matrix.ts:76]: these are already exposed via `export * as Crypto` below (and are documented as such in the jsdoc). Please don't clutter the main namespace more than it has to be.

**Review by richvdh — CHANGES_REQUESTED** [review]

File: src/matrix.ts (hunk 1 above)
Comment (richvdh) [reply src/matrix.ts:76]: The "rule", at least in my head, is that crypto-related stuff should be contained within the `Crypto` namespace. `CryptoEvent` predates my recent work in this area so hasn't (yet) been moved. (And if it was moved, we'd still have to export it at the top level for backwards compat.)

Comment (t3chguy) [inline src/matrix.ts:76]:
```suggestion
```

**Review by richvdh — APPROVED** [review]
thanks!

**Pull request merged**